### PR TITLE
Refactor iterator and optimize performance

### PR DESCRIPTION
**Pull request by ZuoTiJia**
# Required checklist
- [x] Sample config files updated (`config`,`meta/config` and `default config`) 
- [x] If there are user-facing changes, the documentation needs to be updated prior to approving the PR( [Link]() )
- [x] If there are any breaking changes to public APIs, please add the `api change` label.
- [x] Signed [CLA](https://cla-assistant.io/cnosdb/cnosdb) (if not already signed)

# Which issue does this PR close?

Related 

# Rationale for this change

### CODE REVIEW

File: tskv/src/reader/iterator.rs [hunk 1]
```diff
@@ -396,13 +399,222 @@ pub struct FieldFileLocation {
...
-    /// The first index of a DataType in a DataBlock
-    data_block_i: usize,
-    /// The last index of a DataType in a DataBlock
-    data_block_i_end: usize,
-    intersected_time_ranges: TimeRanges,
-    intersected_time_ranges_i: usize,
+    data_block_reader: DataBlockReader,
+}
+
+struct DataTypeWithFileId {
```
Comment (roseboy-liu) [inline tskv/src/reader/iterator.rs:405]: what is the difference between the original design

**Review by roseboy-liu — CHANGES_REQUESTED** [review]

File: tskv/src/reader/iterator.rs (hunk 1 above)
Comment (ZuoTiJia) [reply tskv/src/reader/iterator.rs:405]: There is no difference, it just becomes easier to read.

File: common/models/src/predicate/domain.rs [hunk 2]
```diff
@@ -242,12 +242,8 @@ impl TimeRanges {
     }
 
     pub fn overlaps(&self, time_range: &TimeRange) -> bool {
-        for tr in self.inner.iter() {
-            if tr.overlaps(time_range) {
-                return true;
-            }
-        }
-        false
+        self.max_time_range().overlaps(time_range)
```
Comment (roseboy-liu) [inline common/models/src/predicate/domain.rs:245]: break origin logic.

**Review by roseboy-liu — CHANGES_REQUESTED** [review]

**Comment by yukkit** [comment]
begin benchmark

**Comment by roseboy-liu** [comment]
https://cnosdb-admin.teleport.cnosdb.com/benchmark-hits/hits_87ef0e5.html
https://cnosdb-admin.teleport.cnosdb.com/benchmark-tsbs/tsbs_87ef0e5.html

**Comment by yukkit** [comment]
begin benchmark

**Comment by roseboy-liu** [comment]
https://cnosdb-admin.teleport.cnosdb.com/benchmark-hits/hits_87ef0e5.html
https://cnosdb-admin.teleport.cnosdb.com/benchmark-tsbs/tsbs_87ef0e5.html

**Comment by yukkit** [comment]
begin benchmark

**Comment by roseboy-liu** [comment]
https://cnosdb-admin.teleport.cnosdb.com/benchmark-hits/hits_87ef0e5.html
https://cnosdb-admin.teleport.cnosdb.com/benchmark-tsbs/tsbs_87ef0e5.html

**Review by roseboy-liu — APPROVED** [review]

File: tskv/src/tseries_family.rs [hunk 3]
```diff
@@ -293,7 +296,7 @@ impl LevelInfo {
         self.time_range = TimeRange::new(min_ts, max_ts);
     }
 
-    pub async fn read_column_file(
+    pub(self) async fn read_column_file(
```
[the remark this answers is not in the document]
Comment (ZuoTiJia) [reply tskv/src/tseries_family.rs:299]: This just controls the visibility of the function.

**Pull request merged**